<span style="font-family: Arial; font-weight:bold;font-size:1.5em;">Problem Statement

Employee Attrition has always been a crucial concern for a company. Attrition means a loss to the company in terms of cost invested in training an employee. There are a few companies which were forced to close due to high level of attrition. To minimize attrition several companies tries to analyze and understand several factors which migh lead to an employee to leave the company. 
The IBM HR Attrition Case Study is a dataset which aims to identify important factors that might be influential in determining an employee who might leave the company. You are hired by the company and your objective is to undestand the data and to identify the patterns in the data which will help HR team to take decisive actions to reduce the attrition rate.

There are two given dataset, Company_Data.csv contains information of an employee related to the company and Personal_Data cotains personal information of an employee.

## Data Dictionary

1. EmpId-              Employee ID
2. AGE-                Employee Age
3. EducationField-     Field of Education
4. Gender        -     Employee Gender
5. MaritalStatus -     Employee Marital Status
6. DistanceFromHome-   Distance from home to work (in miles)
7. BusinessTravel-     Frequency of business travel
8. Department-         Department of the employee 
9. HourlyRate-         Hourly rate of an employee
10. JobLevel-          Job Level of the employee. Higher job level means senior post
11. JobRole-           Job Role of an employee
12. JobSatisfaction-   Job Satisfaction level of  an employee
13. MonthlyRate-       Montly rate of an employee
14. NumCompaniesWorked-How many companies has the employee worked before
15. OverTime-          Does the employee gets overtime
16. PercentSalaryHike- Average percentage take for the employee
17. TotalWorkingYears- Total Years of Experience of an employee
18. WorkLifeBalance-   Work life balance of an employee (4 means highly satisfied)
19. YearsAtCompany-    Years employee have worked in current company
20. YearsInCurrentRole-Years employee have worked in current role
21. TotalWorkingYears- Total Years of Experience of an employee
22. YearsSinceLastPromotion-   Number of years since last promotion
23. YearsWithCurrManager- Number of years with current manager
24. Attrition-         Employee has left the company or not


### Change your Working Directory

In [1]:
import os
#os.chdir("")

### Q1. Import all the necessary libraries (Numpy and Pandas)

In [3]:
import numpy as np
import pandas as pd

### Q2. Import both csv files (and name Company_Data as "company" and Personal_Data as "personal" )

Load the Data file into Python DataFrame using pandas read_csv method

In [21]:
company = pd.read_csv("Company_Data.csv")
personal = pd.read_csv("Personal_Data.csv")

### Q3.  View Top 10 rows of both the datasets

Hint:Head function is used to view the top records. The number of records to be viewed, needs to be  given in the parenthesis. Default value is 5.

In [22]:
company.head(10)

EmpID     BusinessTravel              Department  HourlyRate  JobLevel  \
0  2102210      Travel_Rarely                   Sales          94       2.0   
1  2102211  Travel_Frequently  Research & Development          61       2.0   
2  2102212      Travel_Rarely  Research & Development          92       1.0   
3  2102213  Travel_Frequently  Research & Development          56       1.0   
4  2102214      Travel_Rarely  Research & Development          40       1.0   
5  2102215  Travel_Frequently  Research & Development          79       1.0   
6  2102216      Travel_Rarely  Research & Development          81       1.0   
7  2102217      Travel_Rarely  Research & Development          67       1.0   
8  2102218  Travel_Frequently  Research & Development          44       3.0   
9  2102219      Travel_Rarely  Research & Development          94       2.0   

                     JobRole  JobSatisfaction  MonthlyRate  \
0            Sales Executive                4        19479   
1         Research Scientist                2        24907   
2      Laboratory Technician                3         2396   
3         Research Scientist                3        23159   
4      Laboratory Technician                2        16632   
5      Laboratory Technician                4        11864   
6      Laboratory Technician                1         9964   
7      Laboratory Technician                3        13335   
8     Manufacturing Director                3         8787   
9  Healthcare Representative                3        16577   

   NumCompaniesWorked OverTime  PercentSalaryHike  TotalWorkingYears  \
0                   8      Yes                 11                  8   
1                   1       No                 23                 10   
2                   6      Yes                 15                  7   
3                   1      Yes                 11                  8   
4                   9       No                 12                  6   
5                   0       No                 13                  8   
6                   4      Yes                 20                 12   
7                   1       No                 22                  1   
8                   0       No                 21                 10   
9                   6       No                 13                 17   

   WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
0              1.0               6                   4   
1              3.0              10                   7   
2              3.0               0                   0   
3              3.0               8                   7   
4              3.0               2                   2   
5              2.0               7                   7   
6              2.0               1                   0   
7              3.0               1                   0   
8              3.0               9                   7   
9              2.0               7                   7   

   YearsSinceLastPromotion  YearsWithCurrManager Attrition  
0                        0                     5       Yes  
1                        1                     7        No  
2                        0                     0       Yes  
3                        3                     0        No  
4                        2                     2        No  
5                        3                     6        No  
6                        0                     0        No  
7                        0                     0        No  
8                        1                     8        No  
9                        7                     7        No

In [23]:
personal.head(10)

EmpID  Age EducationField  Gender MaritalStatus  DistanceFromHome
0  2102210   41  Life Sciences  Female        Single                 1
1  2102211   49  Life Sciences    Male       Married                 8
2  2102212   37          Other    Male        Single                 2
3  2102213   33  Life Sciences  Female       Married                 3
4  2102214   27        Medical    Male       Married                 2
5  2102215   32  Life Sciences    Male        Single                 2
6  2102216   59        Medical  Female       Married                 3
7  2102217   30  Life Sciences    Male      Divorced                24
8  2102218   38  Life Sciences    Male        Single                23
9  2102219   36        Medical    Male       Married                27

### Q4. Merge both datasets and name the merged dataset as df

Hint:Merge the datasets using pd.merge() command- Use how="outer" and on="EmpID"

In [24]:
df = pd.merge(company,personal,how="outer",on="EmpID")

In [25]:
df

EmpID     BusinessTravel              Department  HourlyRate  \
0     2102210      Travel_Rarely                   Sales          94   
1     2102211  Travel_Frequently  Research & Development          61   
2     2102212      Travel_Rarely  Research & Development          92   
3     2102213  Travel_Frequently  Research & Development          56   
4     2102214      Travel_Rarely  Research & Development          40   
...       ...                ...                     ...         ...   
1465  2103675  Travel_Frequently  Research & Development          41   
1466  2103676      Travel_Rarely  Research & Development          42   
1467  2103677      Travel_Rarely  Research & Development          87   
1468  2103678  Travel_Frequently                   Sales          63   
1469  2103679      Travel_Rarely  Research & Development          82   

      JobLevel                    JobRole  JobSatisfaction  MonthlyRate  \
0          2.0            Sales Executive                4        19479   
1          2.0         Research Scientist                2        24907   
2          1.0      Laboratory Technician                3         2396   
3          1.0         Research Scientist                3        23159   
4          1.0      Laboratory Technician                2        16632   
...        ...                        ...              ...          ...   
1465       2.0      Laboratory Technician                4        12290   
1466       3.0  Healthcare Representative                1        21457   
1467       2.0     Manufacturing Director                2         5174   
1468       2.0            Sales Executive                2        13243   
1469       2.0      Laboratory Technician                3        10228   

      NumCompaniesWorked OverTime  ...  YearsAtCompany  YearsInCurrentRole  \
0                      8      Yes  ...               6                   4   
1                      1       No  ...              10                   7   
2                      6      Yes  ...               0                   0   
3                      1      Yes  ...               8                   7   
4                      9       No  ...               2                   2   
...                  ...      ...  ...             ...                 ...   
1465                   4       No  ...               5                   2   
1466                   4       No  ...               7                   7   
1467                   1      Yes  ...               6                   2   
1468                   2       No  ...               9                   6   
1469                   2       No  ...               4                   3   

      YearsSinceLastPromotion  YearsWithCurrManager  Attrition  Age  \
0                           0                     5        Yes   41   
1                           1                     7         No   49   
2                           0                     0        Yes   37   
3                           3                     0         No   33   
4                           2                     2         No   27   
...                       ...                   ...        ...  ...   
1465                        0                     3         No   36   
1466                        1                     7         No   39   
1467                        0                     3         No   27   
1468                        0                     8         No   49   
1469                        1                     2         No   34   

      EducationField  Gender  MaritalStatus DistanceFromHome  
0      Life Sciences  Female         Single                1  
1      Life Sciences    Male        Married                8  
2              Other    Male         Single                2  
3      Life Sciences  Female        Married                3  
4            Medical    Male        Married                2  
...              ...     ...            ...              ...  
1465       

 ### Q5. View Last 5 records of combined dataframe "df".

Hint:Tail function is used to view the last records. The number of records to be viewed, needs to be given in the parenthesis.Default value is 5.

In [26]:
df.tail()

EmpID     BusinessTravel              Department  HourlyRate  \
1465  2103675  Travel_Frequently  Research & Development          41   
1466  2103676      Travel_Rarely  Research & Development          42   
1467  2103677      Travel_Rarely  Research & Development          87   
1468  2103678  Travel_Frequently                   Sales          63   
1469  2103679      Travel_Rarely  Research & Development          82   

      JobLevel                    JobRole  JobSatisfaction  MonthlyRate  \
1465       2.0      Laboratory Technician                4        12290   
1466       3.0  Healthcare Representative                1        21457   
1467       2.0     Manufacturing Director                2         5174   
1468       2.0            Sales Executive                2        13243   
1469       2.0      Laboratory Technician                3        10228   

      NumCompaniesWorked OverTime  ...  YearsAtCompany  YearsInCurrentRole  \
1465                   4       No  ...               5                   2   
1466                   4       No  ...               7                   7   
1467                   1      Yes  ...               6                   2   
1468                   2       No  ...               9                   6   
1469                   2       No  ...               4                   3   

      YearsSinceLastPromotion  YearsWithCurrManager  Attrition  Age  \
1465                        0                     3         No   36   
1466                        1                     7         No   39   
1467                        0                     3         No   27   
1468                        0                     8         No   49   
1469                        1                     2         No   34   

      EducationField Gender  MaritalStatus DistanceFromHome  
1465         Medical   Male        Married               23  
1466         Medical   Male        Married                6  
1467   Life Sciences   Male        Married                4  
1468         Medical   Male        Married                2  
1469         Medical   Male        Married                8  

[5 rows x 23 columns]

### Q6. Display the shape/dimensions of dataset.

Displaying the shape of the Data Frame in which first value gives number of Rows and second value gives number of columns.

Hint: Use "shape" function 

In [27]:
df.shape

(1470, 23)

In [31]:
print("The number of rows are:",df.shape[0]," columns are:",df.shape[1])

The number of rows are: 1470  columns are: 23


### Q7. Check for missing values. Display number of null values per column.

Hint: use isnull() function

In [33]:
df.isnull().sum()

EmpID                      0
BusinessTravel             0
Department                 0
HourlyRate                 0
JobLevel                   6
JobRole                    0
JobSatisfaction            0
MonthlyRate                0
NumCompaniesWorked         0
OverTime                   0
PercentSalaryHike          0
TotalWorkingYears          0
WorkLifeBalance            4
YearsAtCompany             0
YearsInCurrentRole         0
YearsSinceLastPromotion    0
YearsWithCurrManager       0
Attrition                  0
Age                        0
EducationField             0
Gender                     0
MaritalStatus              0
DistanceFromHome           0
dtype: int64

There are missing values present in the "Joblevel" and "WorklifeBalance" variables of the dataset.

### Q8.  Drop null values from "df" and check if there are any missing value left after dropping null values

In [34]:
df.dropna(inplace=True)

In [35]:
df.isnull().sum()

EmpID                      0
BusinessTravel             0
Department                 0
HourlyRate                 0
JobLevel                   0
JobRole                    0
JobSatisfaction            0
MonthlyRate                0
NumCompaniesWorked         0
OverTime                   0
PercentSalaryHike          0
TotalWorkingYears          0
WorkLifeBalance            0
YearsAtCompany             0
YearsInCurrentRole         0
YearsSinceLastPromotion    0
YearsWithCurrManager       0
Attrition                  0
Age                        0
EducationField             0
Gender                     0
MaritalStatus              0
DistanceFromHome           0
dtype: int64

### Q9. Check the datatype, number of non negative values and name of each variable in the dataset.

Hint : Use info() function

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1460 entries, 0 to 1469
Data columns (total 23 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   EmpID                    1460 non-null   int64  
 1   BusinessTravel           1460 non-null   object 
 2   Department               1460 non-null   object 
 3   HourlyRate               1460 non-null   int64  
 4   JobLevel                 1460 non-null   float64
 5   JobRole                  1460 non-null   object 
 6   JobSatisfaction          1460 non-null   int64  
 7   MonthlyRate              1460 non-null   int64  
 8   NumCompaniesWorked       1460 non-null   int64  
 9   OverTime                 1460 non-null   object 
 10  PercentSalaryHike        1460 non-null   int64  
 11  TotalWorkingYears        1460 non-null   int64  
 12  WorkLifeBalance          1460 non-null   float64
 13  YearsAtCompany           1460 non-null   int64  
 14  YearsInCurrentRole      

### Q10. Check the summary of the dataset
Hint: To see the summary of a dataframe we will use describe function.

In [41]:
df.describe(include="all")

EmpID BusinessTravel              Department   HourlyRate  \
count   1.460000e+03           1460                    1460  1460.000000   
unique           NaN              3                       3          NaN   
top              NaN  Travel_Rarely  Research & Development          NaN   
freq             NaN           1036                     952          NaN   
mean    2.102949e+06            NaN                     NaN    65.914384   
std     4.220800e+02            NaN                     NaN    20.316655   
min     2.102210e+06            NaN                     NaN    30.000000   
25%     2.102585e+06            NaN                     NaN    48.000000   
50%     2.102950e+06            NaN                     NaN    66.000000   
75%     2.103314e+06            NaN                     NaN    84.000000   
max     2.103679e+06            NaN                     NaN   100.000000   

           JobLevel          JobRole  JobSatisfaction   MonthlyRate  \
count   1460.000000             1460      1460.000000   1460.000000   
unique          NaN                9              NaN           NaN   
top             NaN  Sales Executive              NaN           NaN   
freq            NaN              325              NaN           NaN   
mean       2.065753              NaN         2.729452  14299.465068   
std        1.107544              NaN         1.103963   7123.418042   
min        1.000000              NaN         1.000000   2094.000000   
25%        1.000000              NaN         2.000000   8039.750000   
50%        2.000000              NaN         3.000000  14225.500000   
75%        3.000000              NaN         4.000000  20460.500000   
max        5.000000              NaN         4.000000  26999.000000   

        NumCompaniesWorked OverTime  ...  YearsAtCompany  YearsInCurrentRole  \
count          1460.000000     1460  ...     1460.000000         1460.000000   
unique                 NaN        2  ...             NaN                 NaN   
top                    NaN       No  ...             NaN                 NaN   
freq                   NaN     1048  ...             NaN                 NaN   
mean              2.696575      NaN  ...        7.028767            4.243836   
std               2.499154      NaN  ...        6.139390            3.628309   
min               0.000000      NaN  ...        0.000000            0.000000   
25%               1.000000      NaN  ...        3.000000            2.000000   
50%               2.000000      NaN  ...        5.000000            3.000000   
75%               4.000000      NaN  ...       10.000000            7.000000   
max               9.000000      NaN  ...       40.000000           18.000000   

        YearsSinceLastPromotion  YearsWithCurrManager  Attrition          Age  \
count               1460.000000           1460.000000       1460  1460.000000   
unique                      NaN                   NaN          2          NaN   
top                         NaN                   NaN         No          NaN   
freq                        NaN                   NaN       1225          NaN   
mean                   2.197945              4.133562        NaN    36.910959   
std                    3.229596              3.573852        NaN     9.126498   
min                    0.000000              0.000000        NaN    18.000000   
25%                    0.000000              2.000000        NaN    30.000000   
50%                    1.000000              3.000000        NaN    36.000000   
75%                    3.000000              7.000000        NaN    43.000000   
max                   15.000000             17.000000        NaN    60.000000   

        EducationField Gender  MaritalStatus DistanceFromHome  
count             1460   1460           1460      1460.000000  
unique               6      2              3              NaN  
top      Life Sciences   Male        Married              NaN  
freq               600    875            671              N

### Q11. Drop  the variable EmpID from the dataframe df.
To drop any variable, use drop() function.

In [45]:
df.drop("EmpID",axis=1,inplace=True)

In [46]:
df

BusinessTravel              Department  HourlyRate  JobLevel  \
0         Travel_Rarely                   Sales          94       2.0   
1     Travel_Frequently  Research & Development          61       2.0   
2         Travel_Rarely  Research & Development          92       1.0   
3     Travel_Frequently  Research & Development          56       1.0   
4         Travel_Rarely  Research & Development          40       1.0   
...                 ...                     ...         ...       ...   
1465  Travel_Frequently  Research & Development          41       2.0   
1466      Travel_Rarely  Research & Development          42       3.0   
1467      Travel_Rarely  Research & Development          87       2.0   
1468  Travel_Frequently                   Sales          63       2.0   
1469      Travel_Rarely  Research & Development          82       2.0   

                        JobRole  JobSatisfaction  MonthlyRate  \
0               Sales Executive                4        19479   
1            Research Scientist                2        24907   
2         Laboratory Technician                3         2396   
3            Research Scientist                3        23159   
4         Laboratory Technician                2        16632   
...                         ...              ...          ...   
1465      Laboratory Technician                4        12290   
1466  Healthcare Representative                1        21457   
1467     Manufacturing Director                2         5174   
1468            Sales Executive                2        13243   
1469      Laboratory Technician                3        10228   

      NumCompaniesWorked OverTime  PercentSalaryHike  ...  YearsAtCompany  \
0                      8      Yes                 11  ...               6   
1                      1       No                 23  ...              10   
2                      6      Yes                 15  ...               0   
3                      1      Yes                 11  ...               8   
4                      9       No                 12  ...               2   
...                  ...      ...                ...  ...             ...   
1465                   4       No                 17  ...               5   
1466                   4       No                 15  ...               7   
1467                   1      Yes                 20  ...               6   
1468                   2       No                 14  ...               9   
1469                   2       No                 12  ...               4   

      YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  \
0                      4                        0                     5   
1                      7                        1                     7   
2                      0                        0                     0   
3                      7                        3                     0   
4                      2                        2                     2   
...                  ...                      ...                   ...   
1465                   2                        0                     3   
1466                   7                        1                     7   
1467                   2                        0                     3   
1468                   6                        0                     8   
1469                   3                        1                     2   

      Attrition  Age EducationField  Gender MaritalStatus DistanceFromHome  
0           Yes   41  Life Sciences  Female        Single                1  
1            No   49  Life Sciences    Male       Married                8  
2           Yes   37          Other    Male        Single                2  
3            No   33  Life Sciences  Female       Married                3  
4            No   27        Medical    Male       Married                2  
...         ...  ...            ...     ...           ...      

### Q12. Check how many employees left the company from the given dataset
Hint :Use value_counts() function

In [58]:
df.Attrition.value_counts()

No     1225
Yes     235
Name: Attrition, dtype: int64

In the given dataset, there are 235 employees who left the company.

In [60]:
df.Attrition.value_counts(normalize=True)

No     0.839041
Yes    0.160959
Name: Attrition, dtype: float64

In the given dataset, 16.12% left the company.

### Q13. Create a subset "df_new" from the dataset "df" which contain data of all the employees who have attrited. Also check the shape of the new dataset

In [66]:
df_new = df[df["Attrition"]=="Yes"]
df_new.head()

BusinessTravel              Department  HourlyRate  JobLevel  \
0       Travel_Rarely                   Sales          94       2.0   
2       Travel_Rarely  Research & Development          92       1.0   
21      Travel_Rarely                   Sales          82       1.0   
24      Travel_Rarely  Research & Development          83       1.0   
26  Travel_Frequently  Research & Development          72       1.0   

                  JobRole  JobSatisfaction  MonthlyRate  NumCompaniesWorked  \
0         Sales Executive                4        19479                   8   
2   Laboratory Technician                3         2396                   6   
21   Sales Representative                1         6986                   7   
24     Research Scientist                1        17102                   2   
26     Research Scientist                1         4681                   1   

   OverTime  PercentSalaryHike  ...  YearsAtCompany  YearsInCurrentRole  \
0       Yes                 11  ...               6                   4   
2       Yes                 15  ...               0                   0   
21       No                 23  ...               5                   3   
24       No                 11  ...               4                   2   
26      Yes                 22  ...              10                   2   

    YearsSinceLastPromotion  YearsWithCurrManager  Attrition  Age  \
0                         0                     5        Yes   41   
2                         0                     0        Yes   37   
21                        0                     3        Yes   36   
24                        1                     3        Yes   34   
26                        6                     7        Yes   32   

   EducationField  Gender MaritalStatus DistanceFromHome  
0   Life Sciences  Female        Single                1  
2           Other    Male        Single                2  
21  Life Sciences    Male        Single                9  
24        Medical    Male        Single                6  
26  Life Sciences  Female        Single               16  

[5 rows x 22 columns]

In [68]:
df_new.shape

(235, 22)

### Q14. Create the top 5 rows and 5 columns in a matrix format. Use dataframe "df"

Hint- use iloc function

In [71]:
df.iloc[0:5,0:5]

BusinessTravel              Department  HourlyRate  JobLevel  \
0      Travel_Rarely                   Sales          94       2.0   
1  Travel_Frequently  Research & Development          61       2.0   
2      Travel_Rarely  Research & Development          92       1.0   
3  Travel_Frequently  Research & Development          56       1.0   
4      Travel_Rarely  Research & Development          40       1.0   

                 JobRole  
0        Sales Executive  
1     Research Scientist  
2  Laboratory Technician  
3     Research Scientist  
4  Laboratory Technician

### Q15.  Analyze the mean for all the variables for employees who have attrited and whose who have not.

Hint: Use groupby() function

In [74]:
df.groupby("Attrition").mean()

HourlyRate  JobLevel  JobSatisfaction   MonthlyRate  \
Attrition                                                        
No          65.977959  2.146939         2.780408  14254.193469   
Yes         65.582979  1.642553         2.463830  14535.455319   

           NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
Attrition                                                             
No                   2.648980          15.242449          11.862857   
Yes                  2.944681          15.093617           8.280851   

           WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
Attrition                                                        
No                2.782041        7.388571            4.497959   
Yes               2.659574        5.153191            2.919149   

           YearsSinceLastPromotion  YearsWithCurrManager        Age  \
Attrition                                                             
No                        2.243265              4.377143  37.542041   
Yes                       1.961702              2.863830  33.621277   

           DistanceFromHome  
Attrition                    
No                 8.902041  
Yes               10.582979

### Q16. Check the details of 5 employees who have spent maximum years in the company 

In [77]:
df.sort_values(by="YearsAtCompany",ascending=False).head()

BusinessTravel              Department  HourlyRate  JobLevel  \
126   Travel_Rarely  Research & Development          94       3.0   
98    Travel_Rarely                   Sales          37       4.0   
1116  Travel_Rarely                   Sales          60       5.0   
270   Travel_Rarely  Research & Development          81       5.0   
561   Travel_Rarely                   Sales          31       4.0   

                        JobRole  JobSatisfaction  MonthlyRate  \
126   Healthcare Representative                4         3465   
98              Sales Executive                3        24409   
1116                    Manager                4        23037   
270                     Manager                1        18938   
561                     Manager                1        10084   

      NumCompaniesWorked OverTime  PercentSalaryHike  ...  YearsAtCompany  \
126                    1       No                 12  ...              40   
98                     0       No                 13  ...              37   
1116                   1       No                 21  ...              36   
270                    0      Yes                 14  ...              36   
561                    1       No                 11  ...              34   

      YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  \
126                   10                       15                     6   
98                    10                        1                     8   
1116                   6                        2                    13   
270                   10                        4                    13   
561                    6                        1                    16   

      Attrition  Age EducationField  Gender MaritalStatus DistanceFromHome  
126         Yes   58        Medical  Female       Married               23  
98           No   58        Medical    Male        Single               10  
1116         No   55      Marketing    Male       Married               26  
270          No   55        Medical    Male        Single                1  
561          No   52      Marketing    Male       Married                3  

[5 rows x 22 columns]

### Q17.  Find out the minimum "Percentage Salary Hike" the company has offered?


In [80]:
df.PercentSalaryHike.min()

11

### Q18. What is the maximum Total Working Years of an employee?

In [84]:
df.TotalWorkingYears.max()

40

In [85]:
df.columns

Index(['BusinessTravel', 'Department', 'HourlyRate', 'JobLevel', 'JobRole',
       'JobSatisfaction', 'MonthlyRate', 'NumCompaniesWorked', 'OverTime',
       'PercentSalaryHike', 'TotalWorkingYears', 'WorkLifeBalance',
       'YearsAtCompany', 'YearsInCurrentRole', 'YearsSinceLastPromotion',
       'YearsWithCurrManager', 'Attrition', 'Age', 'EducationField', 'Gender',
       'MaritalStatus', 'DistanceFromHome'],
      dtype='object')

### Q19. What percentage of employees stay in the company given they have spend 7 years or more with the current manager?

In [94]:
df[df["YearsWithCurrManager"]>7]["Attrition"].value_counts(normalize=True)

No     0.918519
Yes    0.081481
Name: Attrition, dtype: float64

### Q20.  What percentage of employees lives >10 miles from the company?

In [100]:
(df[df["DistanceFromHome"]>10]["DistanceFromHome"].count()/df.shape[0])*100

30.068493150684933

### Q21. Create a copy of the dataset , remove the "Attrition" variable from the new dataset and check the shape of the dataset

In [101]:
df_copy = df.copy()
df_copy.drop("Attrition",axis=1,inplace=True)

In [103]:
df_copy.shape

(1460, 21)

# END